src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptycatchblock/InputEmptyCatchBlockDefault2.java

```diff
@@ -0,0 +1,235 @@
+        } catch (IOException | NullPointerException | ArithmeticException e) { /* This is expected*/
+        }
+    }
+
+    private void foo5() {
+        try {
+            throw new IOException();
+        } catch (IOException | NullPointerException | ArithmeticException e) { // violation
+        }
+    }
+
+    private void foo6() {
+        try {
+            throw new IOException();
+        } catch (IOException expected) { // This is expected
+            int k = 0;
+        }
+    }
+
+    public void testTryCatch()
+    {
+        try {
+            int y=0;
+            int u=8;
+            int e=u-y;
+            return;
+        }
+        catch (Exception e) {
+            System.identityHashCode(e);
+            return;
+        }
+        finally
+        {
+            return;
+        }
+    }
+
+    public void testTryCatch2()
+    {
+        try {
+        }
+        catch (Exception e) { //OK
+            //This is expected
+            /* This is expected */
+            /**This is expected */
+        }
+        finally
+        {
+        }
+    }
+
+    public void testTryCatch3()
+    {
+        try {
+            int y=0;
+            int u=8;
+            int e=u-y;
+        }
+        catch (IllegalArgumentException e) {
+            System.identityHashCode(e); //some comment
+            return;
+        }
+        catch (IllegalStateException ex) {
+                System.identityHashCode(ex);
+                return;
+        }
+    }
+
+    public void testTryCatch4()
+    {
+        int y=0;
+        int u=8;
+        try {
+            int e=u-y;
+        }
+        catch (IllegalArgumentException e) {
+            System.identityHashCode(e);
+            return;
+        }
+    }
+    public void setFormats() {
+        try {
+            int k = 4;
+        } catch (Exception e) {
+            Object k = null;
+            if (k != null)
+                k = "ss";
+            else {
+                return;
+            }
+        }
+    }
+    public void setFormats1() {
+        try {
+            int k = 4;
+        } catch (Exception e) {
+            Object k = null;
+            if (k != null) {
+                k = "ss";
+            } else {
+                return;
+            }
+        }
+    }
+    public void setFormats2() {
+        try {
+            int k = 4;
+        } catch (Exception e) {
+            Object k = null;
+            if (k != null) {
+                k = "ss";
+                return;
+            }
+        }
+    }
+    public void setFormats3() {
+        try {
+            int k = 4;
+        } catch (Exception e) {
+            Object k = null;
+            if (k != null) {
+                k = "ss";
+
+            }
+        }
+    }
+
+    private void some() {
+        try {
+            throw new IOException();
+        } catch (IOException e) { // violation
+            /* ololo
+             * blalba
+             */
+        }
```

Comment:
We have a slight problem here. Mutation survives on adding ```// violation``` comment. How to handle these edge cases?

Reply:
Just a guess,
```java
        } catch (IOException e) { /* violation
             * blalba
             */
        }
```
?

Reply:
Yes, I thought of the same thing. We'll then have to modify our regexp in the parser. Will that be fine?

Reply:
What about `} catch (IOException e) { /* //violation`
?

Reply:
done